.azure-pipelines/generate-beta-typings.yml

```diff
@@ -0,0 +1,166 @@
+    source: (beta - 3) msgraph-publish-cleanmetadata
+    trigger:
+      branches:
+      - master
+
+pool:
+  vmImage: windows-latest # Info about this image: [0][1]
+
+variables:
+  - group: MicrosoftGraph # Variable group, where variables not set here come from. Set in Azure DevOps
+
+steps:
+- checkout: msgraph-typescript-typings
+  clean: true
+  fetchDepth: 1
+  persistCredentials: true
+- checkout: msgraph-metadata
+  clean: true
+  fetchDepth: 1
+
+- task: PowerShell@2 # Setup environment variables and make them available to all tasks. See [1] for more info.
+  displayName: 'Calculate and set pipeline variables for this job'
+  inputs:
+    targetType: inline
+    script: |
+      $repoModelsDir = "$env:Build_SourcesDirectory\msgraph-typescript-typings\"
+      Write-Host "Path to repo model directory: $repoModelsDir"
+      Write-Host "##vso[task.setvariable variable=repoModelsDir]$repoModelsDir"
+
+      $outputPath = Join-Path $env:Build_SourcesDirectory "output"
+      Write-Host "Path to typewriter.exe output $outputPath"
+      Write-Host "##vso[task.setvariable variable=outputPath]$outputPath"
+
+      $cleanMetadata = "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml"
+      Write-Host "Path to clean metadata $cleanMetadata"
+      Write-Host "##vso[task.setvariable variable=cleanMetadata]$cleanMetadata"
+
+      $branchName = "pipelinebuild/beta/$env:Build_BuildId" # Match the spec in the GH Action
+      Write-Host "Branch path spec for the pull request will be $branchName"
+      Write-Host "##vso[task.setvariable variable=branchName]$branchName"
+
+- task: PowerShell@2
+  displayName: 'Git: checkout beta'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      git checkout beta | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: branch from beta named with the build id: $(Build.BuildId)'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      Write-Host "The new branch name will be: $env:branchName"
+      git checkout -B $env:branchName | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: set user config'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      git config user.email "GraphTooling@service.microsoft.com"
+      git config user.name "Microsoft Graph DevX Tooling"
+
+- task: PowerShell@2
+  displayName: 'Remove generated typings from the repo'
+  inputs:
+    targetType: inline
+    script: |
+      $filePath = Join-Path $env:repoModelsDir "microsoft-graph.d.ts"
+      Remove-Item $filePath | Write-Host
+      Write-Host "Removed the existing typings files in the repo." -ForegroundColor Green
```

Comment:
Can be useful for debugging purposes to log the full path of the file that you attempt to remove.